dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/index.test.ts

```diff
@@ -22,11 +23,24 @@ describe('get-utilisation-report-reconciliation-details-by-id.controller', () =>
 
     const getBankNameByIdSpy = jest.spyOn(banksRepo, 'getBankNameById');
 
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
     it('responds with a 404 when the report cannot be found', async () => {
       // Arrange
       const { req, res } = getHttpMocks();
 
-      findOneSpy.mockResolvedValue(null);
+      when(findOneSpy)
+        .calledWith({
+          where: { id: reportId },
+          relations: {
+            feeRecords: {
+              payments: true,
+            },
+          },
+        })
+        .mockResolvedValue(null);
```

Comment:
I didn't think about it like that, but that's a good point. I'll change this so that a random utilisation report gets returned whenever you call this spy in general, and add the `when` assertions for the test cases I want. This should means that if you do call it with something different, it should return a report with an unexpected structure, throwing different errors which I think more accurately captures what we might expect to happen